rust/core-lib/src/movement.rs

```diff
@@ -68,18 +72,10 @@ fn vertical_motion(
     line_delta: isize,
     modify: bool,
 ) -> (usize, Option<HorizPos>) {
-    // The active point of the selection
-    let active = if modify {
-        r.end
-    } else if line_delta < 0 {
-        r.min()
-    } else {
-        r.max()
-    };
-    let col = if let Some(col) = r.horiz { col } else { view.offset_to_line_col(text, active).1 };
+    let (col, line, n_lines) = selection_position(r, view, text, line_delta, modify);
```

Comment:
No change required, but looking at this I think I would personally keep `n_lines` out of this function; it seems pretty separate and it should be a one-liner in both places it's used?

Reply:
Fair, will change.